clcache.py

```diff
@@ -1528,25 +1459,38 @@ def processCompileRequest(cache, compiler, args):
     printTraceStatement("Expanded commandline '{0!s}'".format(cmdLine))
 
     try:
-        sourceFiles, objectFile = CommandLineAnalyzer.analyze(cmdLine)
+        sourceFiles, objectFiles = CommandLineAnalyzer.analyze(cmdLine)
 
-        if len(sourceFiles) > 1:
-            return reinvokePerSourceFile(cmdLine, sourceFiles, environment), '', ''
-        else:
-            assert objectFile is not None
-            if 'CLCACHE_NODIRECT' in os.environ:
-                returnCode, compilerOutput, compilerStderr, cleanupRequired = \
-                    processNoDirect(cache, objectFile, compiler, cmdLine, environment)
-            else:
-                returnCode, compilerOutput, compilerStderr, cleanupRequired = \
-                    processDirect(cache, objectFile, compiler, cmdLine, sourceFiles[0])
-            printTraceStatement("Finished. Exit code {0:d}".format(returnCode))
+        baseCmdLine = []
+        for arg in cmdLine:
+            if arg not in sourceFiles:
+                baseCmdLine.append(arg)
+
+        exitCode = 0
+        cleanupRequired = False
+        with concurrent.futures.ThreadPoolExecutor(max_workers=jobCount(cmdLine)) as executor:
+            jobs = []
+            for i, srcFile in enumerate(sourceFiles):
```

Comment:
Instead of using `enumerate` here, you could use

```python
for srcFile, objectFile in zip(sourceFiles, objectFiles):
```

such that you don't need `objectFiles[i]` further down.